app/views/layouts/_header.html.erb

```diff
@@ -1,12 +1,19 @@
 <header class="navbar">
   <ul class="links">
-    <li><%= link_to "Haxxor-News", "#TODO" %></li>
-    <li><%= link_to "Home", "#TODO" %></li>
-    <li><%= link_to "Submit", "#TODO" %></li>
+    <li><%= link_to "Home", root_path %></li>
+    <li><%= link_to "Submit", new_post_path %></li>
   </ul>
 
   <ul class="auth">
-    <li><%= link_to "Signup", "#TODO" %></li>
-    <li><%= link_to "Login", "#TODO" %></li>
+    
+    <% if logged_in? %> 
+    	<li><%= link_to "Users", users_path %></li>
+      <li><%= link_to "Profile", user_path(current_user) %></li>
+    	<li><%= link_to "Logout", logout_path, method: "delete" %></li>
+   	<% else %>
+   		<li><%= link_to "Login", login_path %></li>
+   		<li><%= link_to "Signup", new_user_path %></li>
+   	<% end %> 
+  </ul>
```

Comment:
Got some funky spacing going on in this view file. Most text editors have packages you can install which manage the consistency of spacing for you, I'd recommend looking into that.